lib/usbd/usbd_transfer.c

```diff
@@ -897,7 +897,7 @@ void *usbd_urb_get_buffer_pointer(usbd_device *dev, usbd_urb *urb, size_t len)
 		return transfer->buffer;
 	}
 
-	return transfer->buffer + transfer->transferred;
+	return (void*)((uintptr_t)transfer->buffer + transfer->transferred);
```

Comment:
You point is valid.
GCC treat void * pointer same as char *, but that is non-standard behaviour.